packages/lib/service/attribute/server/getAttributes.ts

```diff
@@ -228,6 +144,188 @@ export async function getTeamMembersWithAttributeOptionValuePerAttribute({ teamI
   }
 }
 
+function _getAttributeFromAttributeOption({
+  allAttributesOfTheOrg,
+  attributeOptionId,
+}: {
+  allAttributesOfTheOrg: Attribute[];
+  attributeOptionId: AttributeOptionId;
+}) {
+  return allAttributesOfTheOrg.find((attribute) =>
+    attribute.options.some((option) => option.id === attributeOptionId)
+  );
+}
+
+function _getAttributeOptionFromAttributeOption({
+  allAttributesOfTheOrg,
+  attributeOptionId,
+}: {
+  allAttributesOfTheOrg: Attribute[];
+  attributeOptionId: AttributeOptionId;
+}) {
+  const matchingOption = allAttributesOfTheOrg.reduce((found, attribute) => {
+    if (found) return found;
+    return attribute.options.find((option) => option.id === attributeOptionId) || null;
+  }, null as null | (typeof allAttributesOfTheOrg)[number]["options"][number]);
+  return matchingOption;
+}
+
+async function _getOrgMembershipToUserIdForTeam({ orgId, teamId }: { orgId: number; teamId: number }) {
+  const { orgMemberships, teamMemberships } = await MembershipRepository.findMembershipsForBothOrgAndTeam({
+    orgId,
+    teamId,
+  });
+
+  const orgMembershipToUserIdForTeamMembers = new Map<OrgMembershipId, UserId>();
+
+  /**
+   * For an organization with 3000 users and 10 teams, with every team having around 300 members, the total memberships we query from DB are 3000+300 = 3300
+   * So, these are not a lot of records and we could afford to do in memory computations on them.
+   *
+   */
+  teamMemberships.forEach((teamMembership) => {
+    const orgMembership = orgMemberships.find(
+      (orgMembership) => orgMembership.userId === teamMembership.userId
+    );
+    if (!orgMembership) {
+      console.error(
+        `Org membership not found for userId ${teamMembership.userId} in the organization's memberships`
+      );
+      return;
+    }
+    orgMembershipToUserIdForTeamMembers.set(orgMembership.id, orgMembership.userId);
+  });
+
+  return orgMembershipToUserIdForTeamMembers;
+}
+
+async function _queryAllData({ orgId, teamId }: { orgId: number; teamId: number }) {
+  // Get all the attributes with their options first.
+  const [orgMembershipToUserIdForTeamMembers, attributesOfTheOrg] = await Promise.all([
+    _getOrgMembershipToUserIdForTeam({ orgId, teamId }),
+    AttributeRepository.findManyByOrgId({ orgId }),
+  ]);
+
+  const orgMembershipIds = Array.from(orgMembershipToUserIdForTeamMembers.keys());
+
+  // Get all the attributes assigned to the members of the team
+  const attributesToUsersForTeam = await AttributeToUserRepository.findManyByOrgMembershipIds({
+    orgMembershipIds,
+  });
+
+  return {
+    attributesOfTheOrg,
+    attributesToUsersForTeam,
+    orgMembershipToUserIdForTeamMembers,
+  };
+}
+
+async function getAttributesAssignedToMembersOfTeam({ teamId, userId }: { teamId: number; userId?: number }) {
```

Comment:
This is existing fn that will have slow querying time but isn't used in /router flow. It is to be improved later.